.github/workflows/ci.yml

```diff
@@ -131,7 +131,7 @@ jobs:
 
           - name: Install Python dependencies
             run: |
-              pip install --upgrade pip setuptools py
+              pip install --upgrade setuptools py
```

Comment:
What about removing also `setuptools` update? Cannot we simply use the one coming with the system?

Also, is `py` needed there?

Perhaps remove the whole `--upgrade` line...

Reply:
Also, we have the same `pip --upgrade` situation above, see line 106. That might be removed too.